### PR TITLE
IDENTITY-3991 Create a permission for users to approve workflows

### PR DESCRIPTION
**Pull request by cdwijayarathna**

### CODE REVIEW

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java [hunk 1]
```diff
@@ -78,9 +82,48 @@ public UserAdmin() {
...
+        try {
+            org.wso2.carbon.user.api.UserRealm realm = UserMgtDSComponent.getRealmService().getTenantUserRealm
+                    (PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId());
+            AuthorizationManager authorizationManager = realm.getAuthorizationManager();
+
+
+            FlaggedName[] users = getUserAdminProxy().listAllUsers(filter, limit);
+
+            for (int i = 0; i < users.length - 1; i++) {
+                System.out.println(users[i].getItemName());
```
Comment (cdwijayarathna) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java]: removed in https://github.com/wso2/carbon-identity/pull/1330

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java [hunk 2]
```diff
@@ -78,9 +82,48 @@ public UserAdmin() {
      * @throws UserAdminException
      */
     public FlaggedName[] listAllUsers(String filter, int limit) throws UserAdminException {
+
         FlaggedName[] names;
         names = getUserAdminProxy().listAllUsers(filter, limit);
         return names;
+
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java]: unnecessary space

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java [hunk 3]
```diff
@@ -686,6 +761,42 @@ public void addRemoveRolesOfUser(String userName, String[] newRoles, String[] de
...
+            maxLimit)
+            throws UserAdminException {
+
+        List<FlaggedName> permittedUsers = new ArrayList<>();
+        try {
+            org.wso2.carbon.user.api.UserRealm realm = UserMgtDSComponent.getRealmService().getTenantUserRealm
+                    (PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId());
+            AuthorizationManager authorizationManager = realm.getAuthorizationManager();
+
+
```
Comment (pulasthi7) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java]: unnecessary space

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java (hunk 2 above)
Comment (cdwijayarathna) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1675

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java (hunk 3 above)
Comment (cdwijayarathna) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java]: Fixed with https://github.com/wso2/carbon-identity/pull/1675

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/UserAdminClient.java [hunk 4]
```diff
@@ -258,7 +276,22 @@ public void addRemoveUsersOfRole(String roleName, String[] newUsers, String[] de
...
         }
-        
+
+        return new FlaggedName[0];
+    }
+
+    public FlaggedName[] listUserByClaimWithPermission(ClaimValue claimValue, String filter, String permission, int
+            limit) throws AxisFault {
+        try {
+            org.wso2.carbon.user.mgt.stub.types.carbon.ClaimValue newClaimValue = new
```
Comment (Kishanthan) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/UserAdminClient.java]: FQN of the class is used here. Are there other classes with the same name used in this code?

**Pull request merged**